### ISSUE TITLE
Top Navigation Accessibility Issue

### ISSUE THREAD

**Issue by MI6-55** [mi6/ic-ui-kit#1887]
## Summary of the bug 
For the Top Navigation component when the screen is resized and the burger menu containing navigation links is open, the tooltip to close the menu isn't dismissible. 

## 🪜 How to reproduce 
Tell us the steps to reproduce the problem: 
1. Go to page: [Top Navigation](https://design.sis.gov.uk/components/top-navigation/code/#variants)
2. Resize the window so the burger menu become visible
3. Either move the mouse over the 'X' or tab to it - the tooltip becomes visible
4. Try to close the tooltip via the esc key.
5. The whole modal/window closes instead of the tooltip being dismissed.
6. This does not meet [WCAG criteria 1.4.13](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html)

## 📸 Screenshots or code 
![image](https://github.com/mi6/ic-ui-kit/assets/121944166/36cff415-b371-48ec-98da-0a2ba7546127)

## 🖥 📱 Device 
- Type: Desktop
- Browser: Chrome

## 🧐 Expected behaviour 
To meet 1.4.13:
Where receiving and then removing pointer hover or keyboard focus triggers additional content to become visible and then hidden, the following are true:

Dismissible
A [mechanism](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html#dfn-mechanism) is available to dismiss the additional content without moving pointer hover or keyboard focus, unless the additional content communicates an [input error](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html#dfn-input-error) or does not obscure or replace other content;
Hoverable
If pointer hover can trigger the additional content, then the pointer can be moved over the additional content without the additional content disappearing;
Persistent
The additional content remains visible until the hover or focus trigger is removed, the user dismisses it, or its information is no longer valid.

## 📝 Acceptance Criteria  
If relevant, describe in full detail the different interactions and edge cases that the component or patterns needs to fulfil. 

The tooltip should be dismissible without closing the navigation menu. 

## Additional info 
Tell us anything else useful to help us fix or understand the problem.
Test with keyboard to ensure the tooltip can be dismissed subject to the 1.4.13 criteria.
